### PR TITLE
Add data setter for `surface` layers

### PR DESCRIPTION
**Pull request by DragaDoncila**
# Description
Add data setter for `surface` layer. 

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Addresses the rest of #2354

# How has this been tested?

- [x] added test for changing data

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Functionality LGTM! But I think we now use a new `trans._` syntax where we can pass the data as a second  arg, see https://github.com/napari/napari/blob/96d320c82af10db0e45edcb06b11020ea322f966/docs/guides/translations.md#additional-arguments-inside-strings.

This should fix the failing localization test

File: napari/layers/surface/surface.py [hunk 1]
```diff
@@ -150,9 +151,11 @@ def __init__(
...
         if len(data) not in (2, 3):
             raise ValueError(
-                'Surface data tuple must be 2 or 3, specifying'
-                'verictes, faces, and optionally vertex values,'
-                f'instead got length {len(data)}.'
+                trans._(
+                    'Surface data tuple must be 2 or 3, specifying'
+                    'vertices, faces, and optionally vertex values,'
+                    'instead got length {}.'.format(len(data))
+                )
```
Comment (goanpeca) [inline napari/layers/surface/surface.py:160]: This should be

```python
trans._(
    'Surface data tuple must be 2 or 3, specifying'
    'vertices, faces, and optionally vertex values,'
    'instead got length {length}.',
    deferred=True,
    length=len(data),
)
```

More examples are available here @DragaDoncila 

https://napari.org/guides/stable/translations.html#deferred-translations

We use now deferred translations when raising errors :)

File: napari/layers/surface/surface.py [hunk 2]
```diff
@@ -190,6 +193,25 @@ def dtype(self):
...
 
+    @data.setter
+    def data(self, data):
+        if len(data) not in (2, 3):
+            raise ValueError(
+                trans._(
+                    'Surface data tuple must be 2 or 3, specifying'
+                    'vertices, faces, and optionally vertex values,'
+                    'instead got length {}.'.format(len(data))
+                )
```
Comment (goanpeca) [inline napari/layers/surface/surface.py:208]: This should be

```python
trans._(
    'Surface data tuple must be 2 or 3, specifying'
    'vertices, faces, and optionally vertex values,'
    'instead got length {length}.',
    deferred=True,
    length=len(data),
)
```

Same here :)

**Review by goanpeca — CHANGES_REQUESTED** [review]
Added some comments on how we need to make `trans` use. For raising errors or warning we now use deferred translations.

https://napari.org/guides/stable/translations.html#deferred-translations

Also, we now are able to pass the kwargs to interpolated directly in trans, so no need to use format anymore.

**Comment by DragaDoncila** [comment]
@goanpeca thanks for clarifying! I was just looking at that PR - very nice!

**Review by goanpeca — APPROVED** [review]
🚀

File: napari/layers/surface/surface.py [hunk 3]
```diff
@@ -190,6 +195,27 @@ def dtype(self):
...
+            )
+        self._vertices = data[0]
+        self._faces = data[1]
+        if len(data) == 3:
+            self._vertex_values = data[2]
+        else:
+            self._vertex_values = np.ones(len(self._vertices))
+
+        self._update_dims()
```
Comment (kevinyamauchi) [inline napari/layers/surface/surface.py:217]: Should we emit a data event in the data setter? I think we do this in the other layers.

**Review by kevinyamauchi — COMMENTED** [review]
Amazing! I was just wishing the `Surface` layer had a `data` setter last night. Thanks!

**Comment by DragaDoncila** [comment]
@kevinyamauchi glad you like it! I think you're right, we probably should. Good catch!

**Pull request merged**